src/controllers/account/account.ts

```diff
@@ -17,35 +31,99 @@ export const generateResetToken = async (): Promise<ResetToken> => {
   return Buffer.concat([input.selector, input.verifier]).toString('base64')
 }
 
-export const getVerifierHash = async (input: ResetToken): Promise<Buffer> => {
+export const getVerifierHash = async (input: AuthToken): Promise<Buffer> => {
   const verifierHash = Buffer.alloc(sodium.crypto_generichash_BYTES)
   sodium.crypto_generichash(verifierHash, input.verifier)
   return verifierHash
 }
 
-// public static async storeResetToken (selector: Buffer, verifierHash: Buffer) {
-//   // TODO store selector in DB along with hash(verifier)
-// }
+export const storeAuthToken = async (email: string, role: string, selector: Buffer, verifierHash: Buffer, longterm: boolean) => {
+  // expiration will be UTC time since epoch of when this token expires
+  let expiration: number
+  const now = new Date()
+  const utcNow = Date.UTC(now.getUTCFullYear(),now.getUTCMonth(), now.getUTCDate() , 
+      now.getUTCHours(), now.getUTCMinutes(), now.getUTCSeconds(), now.getUTCMilliseconds())
+  if (longterm) {
+    expiration = utcNow + MILISECONDS_MONTH
+  } else {
+    expiration = utcNow + MILISECONDS_15_MINUTES
+  }
 
-export const splitSelectorVerifier = async (token: string): Promise<ResetToken> => {
+  // Store email and selector in dynamo DB instance along with hash(verifier)
+  const createParams = {
+    Key: { selector: { S: selector } },
+    Item: {
+      email,
+      role,
+      selector,
+      verifierHash,
+      expiration
+    },
+    TableName: TABLE_NAME,
+    ReturnValues: 'ALL_OLD'
+  }
+  try {
+    await docClient.put(createParams).promise()
+  } catch (e) {
+    console.log('here')
+    console.error(e)
+  }
+}
+
+export const splitSelectorVerifier = async (token: string): Promise<AuthToken> => {
   const tokenBuf = Buffer.from(token, 'base64')
   const selector = tokenBuf.slice(0, 16)
   const verifier = tokenBuf.slice(16)
   return { selector, verifier }
 }
 
-// public static async deleteSelector (resetToken: ResetToken) {
-//   // TODO delete resetToken.selector from DB
-// }
+export const deleteSelector = async (authToken: AuthToken) => {
+  // Delete authToken.selector from dynamo db instance
+  const params = {
+    Key: { selector: { S: authToken.selector }}, 
+    TableName: TABLE_NAME
+  }
+  try {
+    await docClient.deleteItem(params).promise()
+  } catch (e) {
+    console.error(e)
+  }
+}
 
-// public static async getStoredVerifierHash (resetToken: ResetToken): Promise<Buffer> {
-//   // TODO lookup resetToken.selector in DB and return stored verifier hash
-//   return Buffer.alloc(0)
-// }
+export const getStoredVerifierHash = async (authToken: AuthToken): Promise<any> => {
+  // Lookup resetToken.selector in DB and return stored verifier hash
+  const params = {
+    Key: { selector: { S: authToken.selector }}, 
+    TableName: TABLE_NAME
+  }
+  try {
+    const token = await docClient.getItem(params).promise()
+    return Promise.resolve(token)
```

Comment:
Assuming this returns the whole row in the DB, not just the verifier hash ala L115, so rename this variable to something more accurate.

Reply:
done

Reply:
still says token

Reply:
done